src/Game/UI/uiFadeProgress.cpp

```diff
@@ -0,0 +1,55 @@
+#include "Game/UI/uiFadeProgress.h"
+#include "Game/UI/uiUtils.h"
+
+namespace uking::ui {
+
+SEAD_SINGLETON_DISPOSER_IMPL(FadeProgress)
+
+void FadeProgress::init() {
+    mActive = false;
+    mProgress = 0.0;
+    mGaugeValue = 0.0;
+    mSteps = 1.0;
+}
+
+void FadeProgress::reset() {
+    init();
+}
+
+void FadeProgress::updateFadeScreen() {
+    if (mActive) {
```

Comment:
```
if (!mActive)
    return;
```
(rationale: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code)